core/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala

```diff
@@ -157,6 +160,35 @@ abstract class AppBase(
   var mlEventLogType = ""
   var pysparkLogFlag = false
 
+  def getOrCreateExecutor(executorId: String, addTime: Long): ExecutorInfoClass = {
+    executorIdToInfo.getOrElseUpdate(executorId, {
+      new ExecutorInfoClass(executorId, addTime)
+    })
+  }
+
+  /**
+   * Retrieves cluster information based on executor nodes.
+   * If executor nodes exist, calculates the number of hosts and total cores,
+   * and extracts executor and driver instance types (databricks only)
+   *
+   * @return
+   */
+  def getClusterInfo: Option[ClusterInfo] = {
```

Comment:
Can we make the "platform" a field in the cluster information? or this should be part of the AppInfo?
If we put it in the `clusterInfo`, then we get the benefit of using it as input or at least detecting it in the user-tools.